express-backend/models/tasks-services.js

```diff
@@ -78,12 +78,18 @@ async function deleteTask(_id) {
 async function updateTask(_id, updated_fields) {
   // search by description, then update
   let task_id = _id;
-  const query = await TaskModel.findOne(
-    { taskName: updated_fields.taskName },
-    "_id",
-  );
-  console.log(query._id, updated_fields.taskName);
-  task_id = query._id;
+  if (
+    task_id === undefined ||
+    task_id === null ||
+    task_id === "" ||
+    task_id === "undefined"
```

Comment:
The code segment above has been addressed in the recent code commit. Most likely the extra extra checks in the conditional are unnecessary for now however, this may change later on when we unit test our code.